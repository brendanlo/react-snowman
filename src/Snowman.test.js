import { render, fireEvent } from "@testing-library/react";
import Snowman from "./Snowman";


describe("test for max wrong guesses", function () {

  it("ends game after 6 wrong guesses", function () {
    const { container } = render(<Snowman />);

    fireEvent.click(container.querySelector("button[value='b']"));
    fireEvent.click(container.querySelector("button[value='c']"));
    fireEvent.click(container.querySelector("button[value='d']"));
    fireEvent.click(container.querySelector("button[value='x']"));
    fireEvent.click(container.querySelector("button[value='y']"));
    fireEvent.click(container.querySelector("button[value='z']"));

    expect(container.querySelector("button[value='b']")).not.toBeInTheDocument();
    expect(container.querySelector(".Snowman-word").textContent).toEqual("apple");
    expect(container.querySelector(".Snowman-lose-msg").textContent).toEqual("You Lose!!");

  })

  it("is a snapshot of correct lose scenario", function () {
    const { container } = render(<Snowman />);

    fireEvent.click(container.querySelector("button[value='b']"));
    fireEvent.click(container.querySelector("button[value='c']"));
    fireEvent.click(container.querySelector("button[value='d']"));
    fireEvent.click(container.querySelector("button[value='x']"));
    fireEvent.click(container.querySelector("button[value='y']"));
    fireEvent.click(container.querySelector("button[value='z']"));

    expect(container).toMatchSnapshot();
  })

})