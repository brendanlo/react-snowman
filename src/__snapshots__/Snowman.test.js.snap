// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test for max wrong guesses is a snapshot of correct lose scenario 1`] = `
<div>
  <div
    class="Snowman"
  >
    <img
      alt="6"
      src="6.png"
    />
    <p
      class="Snowman-wrong-guesses"
    >
       Number wrong: 
      6
       
    </p>
    <p
      class="Snowman-word"
    >
      apple
    </p>
    <p
      class="Snowman-buttons"
    />
    <p
      class="Snowman-lose-msg"
    >
      You Lose!!
    </p>
  </div>
</div>
`;
